cmd/go-tpc/main.go

```diff
@@ -37,23 +38,48 @@ var (
 	globalCtx context.Context
 )
 
+const (
+	unknownDB       = "Unknown database"
+	connectionError = "connection refused"
+	createDBDDL     = "CREATE DATABASE "
+	mysqlDriver     = "mysql"
+)
+
 func closeDB() {
 	if globalDB != nil {
 		globalDB.Close()
 	}
 	globalDB = nil
 }
 
-func openDB() {
+func openDB() error {
 	// TODO: support other drivers
-	dsn := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", user, password, host, port, dbName)
+	var tmpDB *sql.DB
+	ds := fmt.Sprintf("%s:%s@tcp(%s:%d)/", user, password, host, port)
+	dsn := ds + dbName
 	var err error
-	globalDB, err = sql.Open("mysql", dsn)
+	globalDB, err = sql.Open(mysqlDriver, dsn)
 	if err != nil {
-		panic(err)
+		return err
+	}
+	if err := globalDB.Ping(); err != nil {
+		errString := err.Error()
+		if strings.Contains(errString, unknownDB) {
+			tmpDB, _ = sql.Open(mysqlDriver, ds)
+			defer tmpDB.Close()
+			if _, err := tmpDB.Exec(createDBDDL + dbName); err != nil {
+				return fmt.Errorf("failed to create database, err %v\n", err)
+			}
+		} else if strings.Contains(errString, connectionError) {
+			globalDB = nil
```

Comment:
why need to check connectionError here?

Reply:
You mean don’t check the error kind, just set db to nil?

Reply:
could return err directly here?

Reply:
That sounds reasonable. But it will cause a panic if return err directly. Now I think maybe it is better to not do anything, just set db to nil in this case. What do you think?

Reply:
Code updated. PTAL again.

Reply:
OMG, That's my fault, I think the `host` is '' by default. But this error is intolerant for the case that load data directly to DB, and  panic  when `tpccConfig.OutputDir != ""`